packages/backend-core/src/events/publishers/action.ts

```diff
@@ -0,0 +1,34 @@
+import { publishEvent } from "../events"
+import {
+  Event,
+  ActionAutomationStepExecuted,
+  ActionCrudExecuted,
+  ActionAiAgentExecuted,
+} from "@budibase/types"
+
+async function automationStepExecuted(
+  action: ActionAutomationStepExecuted,
+  timestamp?: string | number
+) {
+  await publishEvent(Event.ACTION_AUTOMATION_STEP_EXECUTED, action, timestamp)
+}
+
+async function crudExecuted(
```

Comment:
It's not CRUD it's CUD 😁 Maybe there's some other name that would work like "writeQuery", idk

*Read* does not trigger an action increment.